sonar-java-plugin/src/main/java/org/sonar/plugins/java/JavaSquidSensor.java

```diff
@@ -92,15 +93,15 @@ public void execute(SensorContext context) {
     sonarComponents.saveAnalysisErrors();
   }
 
-  private Iterable<File> getSourceFiles() {
+  private Collection<File> getSourceFiles() {
     return toFile(fs.inputFiles(fs.predicates().and(fs.predicates().hasLanguage(Java.KEY), fs.predicates().hasType(InputFile.Type.MAIN))));
   }
 
-  private Iterable<File> getTestFiles() {
+  private Collection<File> getTestFiles() {
     return toFile(fs.inputFiles(fs.predicates().and(fs.predicates().hasLanguage(Java.KEY), fs.predicates().hasType(InputFile.Type.TEST))));
   }
 
-  private static Iterable<File> toFile(Iterable<InputFile> inputFiles) {
+  private static Collection<File> toFile(Iterable<InputFile> inputFiles) {
     return StreamSupport.stream(inputFiles.spliterator(), false).map(InputFile::file).collect(Collectors.toList());
```

Comment:
this can be simplified with the move to Collection.

Reply:
Not sure to see how

Reply:
Sorry indeed, was confused by the return type. input type has not changed so comment is irrelevant.